### PR TITLE
Unit tests for the SensitiveDataLogger

### PR DESCRIPTION
**Pull request by alexmg**
## Motivation and Context (Why the change? What's the scenario?)

Contributing unit tests for the `SensitiveDataLogger` to cover a recently fixed bug and prevent future regression.

## High level description (Approach, Design)

N/A - unit tests only

### CODE REVIEW

**Comment by alexmg** [comment]
These unit tests provide coverage for the problem raised in #851

**Comment by alexmg** [comment]
@microsoft-github-policy-service agree

File: service/tests/Abstractions.UnitTests/Diagnostics/SensitiveDataLoggerTests.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using Microsoft.KernelMemory.Diagnostics;
+
+namespace Microsoft.KM.Abstractions.UnitTests.Diagnostics;
+
+public sealed class SensitiveDataLoggerTests : IDisposable
+{
+    private const string EnvironmentVariableName = "ASPNETCORE_ENVIRONMENT";
```
Comment (dluc) [inline service/tests/Abstractions.UnitTests/Diagnostics/SensitiveDataLoggerTests.cs:9]:
```suggestion
    private const string EnvironmentVariableName = "ASPNETCORE_ENVIRONMENT";

    [Fact]
    public void ItIsDisabledByDefault()
    {
        // Assert
        Assert.False(SensitiveDataLogger.Enabled);
    }
```

File: service/tests/Abstractions.UnitTests/Diagnostics/SensitiveDataLoggerTests.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+    [Fact]
+    [Trait("Category", "UnitTest")]
+    public void ItCanBeEnabledInDevelopmentEnvironment()
+    {
+        Environment.SetEnvironmentVariable(EnvironmentVariableName, "Development");
+        SensitiveDataLogger.Enabled = true;
+        Assert.True(SensitiveDataLogger.Enabled);
```
Comment (dluc) [inline service/tests/Abstractions.UnitTests/Diagnostics/SensitiveDataLoggerTests.cs:17]:
```suggestion

        // Act
        SensitiveDataLogger.Enabled = true;

        // Assert
        Assert.True(SensitiveDataLogger.Enabled);
```

File: service/tests/Abstractions.UnitTests/Diagnostics/SensitiveDataLoggerTests.cs [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    }
+
+    [Theory]
+    [InlineData("Staging")]
+    [InlineData("Production")]
+    [Trait("Category", "UnitTest")]
+    public void ItCannotBeEnabledInNonDevelopmentEnvironments(string environment)
+    {
+        Environment.SetEnvironmentVariable(EnvironmentVariableName, environment);
+        Assert.Throws<InvalidOperationException>(() => SensitiveDataLogger.Enabled = true);
```
Comment (dluc) [inline service/tests/Abstractions.UnitTests/Diagnostics/SensitiveDataLoggerTests.cs:27]:
```suggestion

        // Arrange
        Environment.SetEnvironmentVariable(EnvironmentVariableName, environment);

        // Act/Assert
        Assert.Throws<InvalidOperationException>(() => SensitiveDataLogger.Enabled = true);
```

File: service/tests/Abstractions.UnitTests/Diagnostics/SensitiveDataLoggerTests.cs [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    [Theory]
+    [InlineData("Development")]
+    [InlineData("Staging")]
+    [InlineData("Production")]
+    [Trait("Category", "UnitTest")]
+    public void ItCanBeDisabledForAllEnvironments(string environment)
+    {
+        Environment.SetEnvironmentVariable(EnvironmentVariableName, environment);
+        SensitiveDataLogger.Enabled = false;
+        Assert.False(SensitiveDataLogger.Enabled);
```
Comment (dluc) [inline service/tests/Abstractions.UnitTests/Diagnostics/SensitiveDataLoggerTests.cs:39]:
```suggestion
        // Arrange
        Environment.SetEnvironmentVariable(EnvironmentVariableName, environment);

        // Act
        SensitiveDataLogger.Enabled = false;

        // Assert
        Assert.False(SensitiveDataLogger.Enabled);
```

**Review by dluc — COMMENTED** [review]
Minor changes for readability and to handle string case.

One thing I would improve is making sure tests do not run in parallel, because they act on shared state, e.g. the static `.Enabled` and the env var. I tried adding an
```
Assert.False(SensitiveDataLogger.Enabled);
```
to all tests, but they would fail (on my machine). Same using a lock, unclear why.

File: service/tests/Abstractions.UnitTests/Diagnostics/SensitiveDataLoggerTests.cs [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    }
+
+    public void Dispose() => Environment.SetEnvironmentVariable(EnvironmentVariableName, null);
```
Comment (dluc) [inline service/tests/Abstractions.UnitTests/Diagnostics/SensitiveDataLoggerTests.cs:42]:
```suggestion
    public void Dispose()
    {
        Environment.SetEnvironmentVariable(EnvironmentVariableName, null);
        SensitiveDataLogger.Enabled = false;
    }
```

Important to reset also `.Enabled`

**Review by dluc — APPROVED** [review]

**Review by dluc — APPROVED** [review]

**Pull request merged**